ij/src/main/kotlin/org/jetbrains/academy/test/system/test/BaseIjTestClass.kt

```diff
@@ -1,8 +1,55 @@
 package org.jetbrains.academy.test.system.test
 
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiRecursiveElementWalkingVisitor
 import com.intellij.testFramework.fixtures.BasePlatformTestCase
+import org.jetbrains.kotlin.psi.KtBlockExpression
+import org.jetbrains.kotlin.psi.KtNamedFunction
+import org.jetbrains.kotlin.psi.KtProperty
 
 /**
  * A Base test class to create tests with PSI without adding intellij dependency into courses directly.
  */
-open class BaseIjTestClass : BasePlatformTestCase()
+open class BaseIjTestClass : BasePlatformTestCase() {
+
+    fun findMethodsWithContent(content: String): List<String> {
+        val methodNames = mutableListOf<String>()
+        var currentMethod = ""
+        ApplicationManager.getApplication().runReadAction<Unit?> {
+            myFixture.file.accept(object : PsiRecursiveElementWalkingVisitor() {
+                override fun visitElement(element: PsiElement) {
+                    when (element) {
+                        is KtNamedFunction -> currentMethod = element.name.toString()
+                        is KtBlockExpression -> {
+                            if (element.text.contains(content)) {
+                                methodNames.add(currentMethod)
+                            }
+                        }
+                    }
+                    super.visitElement(element)
+                }
+            })
+        }
```

Comment:
I think it is better here to collect all KtNamedFunction elements and then to accepts a filter.

```kotlin
/** Extracts [kotlin elements][KtElement] of given type from kotlin related files in project. */
/** Extracts elements of given type from [PsiElement] subtree. */
fun <T : PsiElement> PsiElement.extractElementsOfType(psiElementClass: Class<T>): MutableCollection<T> =
    PsiTreeUtil.collectElementsOfType(this, psiElementClass)

myFixture.file.extractElementsOfType(KtNamedFunction::class.java)
```

Because in your case you can catch some troubles with nested functions